python/ray/serve/deployment.py

```diff
@@ -163,57 +153,38 @@ def __init__(
-        if not (ray_actor_options is None or isinstance(ray_actor_options, dict)):
-            raise TypeError("ray_actor_options must be a dict.")
-        if placement_group_bundles is not None:
-            if not isinstance(placement_group_bundles, list):
-                raise TypeError("placement_group_bundles must be a list.")
-
-            for bundle in placement_group_bundles:
-                if not isinstance(bundle, dict):
-                    raise TypeError(
-                        "placement_group_bundles entries must be "
-                        f"dicts, got {type(bundle)}."
-                    )
-        if not (
-            placement_group_strategy is None
-            or isinstance(placement_group_strategy, str)
-        ):
-            raise TypeError("placement_group_strategy must be a string.")
 
         if is_driver_deployment is True:
-            if config.num_replicas != 1:
+            if deployment_config.num_replicas != 1:
                 raise ValueError("num_replicas should not be set for driver deployment")
-            if config.autoscaling_config:
+            if deployment_config.autoscaling_config:
                 raise ValueError("autoscaling should not be set for driver deployment")
 
-        if init_args is None:
-            init_args = ()
-        if init_kwargs is None:
-            init_kwargs = {}
-
         docs_path = None
         if (
-            inspect.isclass(func_or_class)
-            and hasattr(func_or_class, "__module__")
-            and func_or_class.__module__ == "ray.serve.api"
-            and hasattr(func_or_class, "__fastapi_docs_path__")
+            inspect.isclass(replica_config.deployment_def)
+            and hasattr(replica_config.deployment_def, "__module__")
+            and replica_config.deployment_def.__module__ == "ray.serve.api"
+            and hasattr(replica_config.deployment_def, "__fastapi_docs_path__")
         ):
-            docs_path = func_or_class.__fastapi_docs_path__
+            docs_path = replica_config.deployment_def.__fastapi_docs_path__
 
-        self._func_or_class = func_or_class
         self._name = name
         self._version = version
-        self._config = config
-        self._init_args = init_args
-        self._init_kwargs = init_kwargs
+        self._deployment_config = deployment_config
+        self._replica_config = replica_config
         self._route_prefix = route_prefix
-        self._ray_actor_options = ray_actor_options
-        self._placement_group_bundles = placement_group_bundles
-        self._placement_group_strategy = placement_group_strategy
         self._is_driver_deployment = is_driver_deployment
         self._docs_path = docs_path
 
+    @property
+    def deployment_config(self) -> DeploymentConfig:
+        return self._deployment_config
+
+    @property
+    def replica_config(self) -> ReplicaConfig:
+        return self._replica_config
```

Comment:
`Deployment` is a public API and `ReplicaConfig` / `DeploymentConfig` currently aren't, so let's make these accessors private please